### PR TITLE
Add tests for unixtime conversion functions 

### PR DESCRIPTION
**Pull request by PrayagS**
This PR fixes a minor issue in the conversion function and adds tests for the same functions.

Please let me know if you want any particular test cases to be included.

### CODE REVIEW

File: servers/apps/test/datetime_test.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    while (timestamp <= TEST_END_TIME) {
+        timestamp_to_datetime(timestamp, &datetime);
+        TEST_ASSERT(datetime.second >= 0 && datetime.second < 60);
+        TEST_ASSERT(datetime.minute >= 0 && datetime.minute < 60);
+        TEST_ASSERT(datetime.hour >= 0 && datetime.hour < 24);
+        TEST_ASSERT(datetime.month == 1);
+        TEST_ASSERT(datetime.year == 2020);
+        ++timestamp;
+    }
```
Comment (nuta) [inline servers/apps/test/datetime_test.c:23]: This loop runs 86400 times and I think we don't need to do that. Instead, I recommend add some tests without a loop:

- normal case: converting to/from a common case, say, `2020-11-20T09:16:55`.
- edge / boundary cases: your *"Test for start of epoch time"* is a good example.
- corner case: e.g. Feb 29 in a leap year.

**Review by nuta — CHANGES_REQUESTED** [review]
Thanks for the PR! Can you take a look at my comment?

File: servers/apps/test/datetime_test.c (hunk 1 above)
Comment (PrayagS) [reply servers/apps/test/datetime_test.c]: I modified the tests as you requested. Please have a look again.

**Comment by PrayagS** [comment]
BTW, are we looking to support negative timestamps? The current design doesn't support that.

File: servers/apps/test/datetime_test.c (hunk 1 above)
Comment (nuta) [reply servers/apps/test/datetime_test.c]: Looks good! Thanks!

**Review by nuta — APPROVED** [review]

**Comment by nuta** [comment]
I have two suggestions on `datetime.c` library:

> BTW, are we looking to support negative timestamps? The current design doesn't support that.
Good point. For now, I think we don't need the support for that.

By the way, is `day_of_week` correct? According to python's implementation, their week days are slight different:

```
>>> datetime.datetime.fromtimestamp(1605866400).weekday()
4
>>> datetime.datetime.fromtimestamp(1582979696).weekday()
5
>>> datetime.datetime.fromtimestamp(0).weekday()
3
```

**Comment by PrayagS** [comment]
> By the way, is `day_of_week` correct? According to python's implementation, their week days are slightly different:
> 
> ```
> >>> datetime.datetime.fromtimestamp(1605866400).weekday()
> 4
> >>> datetime.datetime.fromtimestamp(1582979696).weekday()
> 5
> >>> datetime.datetime.fromtimestamp(0).weekday()
> 3
> ```

It's a design choice. The current design numbers Sunday as the first day of the week (`day_of_week = 1`) and then so on. Seems like the python implementation starts numbering from Monday and uses zero indexing so `day_of_week = 0` and then so on.

**Comment by nuta** [comment]
> > By the way, is `day_of_week` correct? According to python's implementation, their week days are slightly different:
> > ```
> > >>> datetime.datetime.fromtimestamp(1605866400).weekday()
> > 4
> > >>> datetime.datetime.fromtimestamp(1582979696).weekday()
> > 5
> > >>> datetime.datetime.fromtimestamp(0).weekday()
> > 3
> > ```
> 
> It's a design choice. The current design numbers Sunday as the first day of the week (`day_of_week = 1`) and then so on. Seems like the python implementation starts numbering from Monday and uses zero indexing so `day_of_week = 0` and then so on.

That makes sense. Thanks for the explanation!
